### PR TITLE
Support GPG (signed data) container in Homebrew Cask

### PR DESCRIPTION
**Pull request by rednoah**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [X] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [X] Have you successfully run `brew tests` with your changes locally?

-----

This PR adds support for GPG as container format (i.e. [GnuPG signed data](https://www.gnupg.org/gph/en/manual/x135.html)) in Homebrew Cask in addition to the current support for GPG detached signatures.

### CODE REVIEW

File: Library/Homebrew/cask/lib/hbc/container/gpg.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+      def extract
+        if (gpg = which("gpg")).nil?
+          raise CaskError, "Expected to find gpg executable. Cask '#{@cask}' must add: depends_on formula: 'gpg'"
+        end
+
+        import_key
+
+        Dir.mktmpdir do |unpack_dir|
+          @command.run!(gpg, args: ["--batch", "--yes", "--output", Pathname(unpack_dir).join(File.basename(@path.basename, ".gpg")), "--decrypt", @path])
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/container/gpg.rb]: `@path.basename(".gpg")` should work here.

File: Library/Homebrew/cask/lib/hbc/container/gpg.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+require "tmpdir"
+
+require "hbc/container/base"
+
+module Hbc
+  class Container
+    class Gpg < Base
+      def self.me?(criteria)
+        criteria.extension(/gpg/n)
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/container/gpg.rb]: I don't think you need `/n` here. Maybe make this `/\Agpg\Z/`.

**Comment by reitermarkus** [comment]
Do you have a real-world Cask I could test this with?

**Comment by rednoah** [comment]
I've tested with this:
```
cask 'filebot' do
  version '4.7.10'
  sha256 :no_check

  url "https://sourceforge.net/projects/filebot/files/filebot/HEAD/FileBot_#{version}-darwin.tar.xz.gpg"
  gpg :embedded, key_url: 'https://raw.githubusercontent.com/filebot/filebot/master/installer/gpg/maintainer.pub'

  name 'FileBot'
  binary 'FileBot.app/Contents/MacOS/filebot.sh', target: 'filebot'
end
```

**Comment by rednoah** [comment]
I've tested with this:
```
cask 'filebot' do
  version '4.7.10'
  sha256 :no_check

  url "https://sourceforge.net/projects/filebot/files/filebot/HEAD/FileBot_#{version}-darwin.tar.xz.gpg"
  gpg :embedded, key_url: 'https://raw.githubusercontent.com/filebot/filebot/master/installer/gpg/maintainer.pub'

  name 'FileBot'
  binary 'FileBot.app/Contents/MacOS/filebot.sh', target: 'filebot'
end
```

File: Library/Homebrew/cask/lib/hbc/container/gpg.rb (hunk 1 above)
Comment (rednoah) [inline Library/Homebrew/cask/lib/hbc/container/gpg.rb]: I'll try. Mainly copied that one from the other container implementations.

File: Library/Homebrew/cask/lib/hbc/container/gpg.rb (hunk 2 above)
Comment (rednoah) [inline Library/Homebrew/cask/lib/hbc/container/gpg.rb]: OK. Haven't seen the ^...$ usage in the other containers though.

**Comment by rednoah** [comment]
There seem to be issues on the `CI` with running the test suite:
```
Error: stack level too deep
```

**Comment by reitermarkus** [comment]
Thanks!

**Comment by rednoah** [comment]
You're welcome. Thanks for accepting the feature.

**Pull request merged**